src/Sentry/Internal/Http/CachingTransport.cs

```diff
@@ -178,6 +187,48 @@ private async Task CachedTransportBackgroundTaskAsync()
         _options.LogDebug("CachingTransport worker stopped.");
     }
 
+    internal void SalvageAbandonedCacheSessions(CancellationToken cancellationToken)
+    {
+        if (_options.GetBaseCacheDirectoryPath() is not { } baseCacheDir)
+        {
+            return;
+        }
+
+        const string searchPattern = $"{CacheDirectoryHelper.IsolatedCacheDirectoryPrefix}*";
+        foreach (var dir in _fileSystem.EnumerateDirectories(baseCacheDir, searchPattern))
+        {
+            if (cancellationToken.IsCancellationRequested)
+            {
+                return; // graceful exit on cancellation
+            }
+
+            if (string.Equals(dir, _isolatedCacheDirectoryPath, StringComparison.OrdinalIgnoreCase))
+            {
+                continue; // Ignore the current cache directory
+            }
+
+            _options.LogDebug("found abandoned cache: {0}", dir);
+            using var coordinator = new CacheDirectoryCoordinator(dir, _options.DiagnosticLogger, _options.FileSystem);
+            if (!coordinator.TryAcquire())
```

Comment:
question: Could we run into a concurrency issue?

E.g.
- this recycler has started and tries to acquire a lock-file
- in the `.ctor` we have also acquired a lock-file already, in a directory below
- now we try to move files that contain a lock-file that we still have a reference via `Stream` on

Essentially now retrying 3 times without any effect?
But I may misunderstand something there 🤔